### PR TITLE
Add GAPButton example

### PR DESCRIPTION
**Pull request by LiyouZhou**

### CODE REVIEW

File: BLE_GAPButton/module.json [hunk 1]
```diff
@@ -0,0 +1,12 @@
+{
+  "name": "ble-gapbutton",
+  "version": "0.0.0",
```
Comment (rgrover) [inline BLE_GAPButton/module.json]: please change this to v0.0.1

File: BLE_GAPButton/module.json [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+  "bin": "./source",
+  "description": "BLE button that uses gap advertisement to broadcast its state.",
+  "author": "Liyou Zhou",
+  "license": "Apache-2.0",
+  "dependencies": {
+    "mbed-drivers": "",
+    "ble": "^2.0.3"
```
Comment (rgrover) [inline BLE_GAPButton/module.json]: ^2.0.0 would suffice; unless you require >= 2.0.3

**Comment by rgrover** [comment]
looks very good otherwise. Please make the minor changes requested, and send a review request.

**Comment by LiyouZhou** [comment]
@rgrover Please review.

File: BLE_GAPButton/source/main.cpp [hunk 3]
```diff
@@ -0,0 +1,177 @@
...
+{
+    // Update the count in the SERVICE_DATA field of the advertising payload
+    uint8_t service_data[3];
+    service_data[0] = GAPButtonUUID & 0xff;
+    service_data[1] = GAPButtonUUID >> 8;
+    service_data[2] = count; // Put the button click count in the third byte
+    ble_error_t err = BLE::Instance().gap().updateAdvertisingPayload(GapAdvertisingData::SERVICE_DATA, (uint8_t *)service_data, sizeof(service_data));
+    if (err != BLE_ERROR_NONE) {
+        print_error(err, "Updating payload failed");
+        return;
```
Comment (PrzemekWirkus) [inline BLE_GAPButton/source/main.cpp]: Not needed line.

File: BLE_GAPButton/source/main.cpp [hunk 4]
```diff
@@ -0,0 +1,177 @@
...
+    }
+}
+
+void buttonPressedCallback(void)
+{
+    count++;
```
Comment (PrzemekWirkus) [inline BLE_GAPButton/source/main.cpp]:
```c
++count;
```
will be faster.

Comment (rgrover) [inline BLE_GAPButton/source/main.cpp]: :)

**Comment by rgrover** [comment]
@LiyouZhou : please include @PrzemekWirkus 's recommendations and resubmit. thanks. :)

**Comment by LiyouZhou** [comment]
@rgrover please review.

**Pull request merged**